src/main.c

```diff
@@ -34,12 +34,12 @@ static void write_null(void) {
 #define BS 128
 #define USE_SLURPIN 1
 
-static char *stdin_slurp(int *sz) {
+static char *file_slurp(FILE *f, int *sz) {
```

Comment:
as long as its taking a stream/file-handler as argument,its no longer reading from a file. so better rename this to `slurp`. also sz should be size_t